pkg/util/file_indexer.go

```diff
@@ -96,20 +96,20 @@ func gitIgnoreFilePath(directory string) (string, error) {
 }
 
 // addOdoFileIndex adds odo-file-index.json to .gitignore
-func addOdoFileIndex(ignoreFile string) error {
+func addOdoFileIndex(gitIgnoreFile string) error {
 	var data []byte
-	file, err := os.OpenFile(ignoreFile, os.O_APPEND|os.O_RDWR, 0600)
+	file, err := os.OpenFile(gitIgnoreFile, os.O_APPEND|os.O_RDWR, 0600)
 	if err != nil {
 		return errors.Wrap(err, "failed to open .gitignore file")
 	}
 	defer file.Close()
 
-	if data, err = ioutil.ReadFile(ignoreFile); err != nil {
+	if data, err = ioutil.ReadFile(gitIgnoreFile); err != nil {
 		return errors.Wrap(err, "failed reading data from .gitignore file")
 	}
 	// check whether .odo/odo-file-index.json is already in the .gitignore file
-	if !strings.Contains(string(data), ".odo/odo-file-index.json") {
-		if _, err := file.WriteString("\n" + ".odo/odo-file-index.json"); err != nil {
+	if !strings.Contains(string(data), filepath.Join(fileIndexDirectory, fileIndexName)) {
+		if _, err := file.WriteString("\n" + filepath.Join(fileIndexDirectory, fileIndexName)); err != nil {
 			return errors.Wrap(err, "failed to Add odo-file-index.json to .gitignore file")
```

Comment:
You mean `"failed to add %v to .gitignore file", fileIndexName`? :smile:

Reply:
Yes :-D